Assets/MixedRealityToolkit.ThirdParty/MRTK-Quest/Scripts/Input/Controllers/OculusQuestHand.cs

```diff
@@ -337,13 +341,40 @@ protected bool UpdateHandData(OVRHand ovrHand, OVRSkeleton ovrSkeleton)
                 }
             }
 
+            // Pinch was also used as grab, we want to allow hand-curl grab not just pinch.
+            // Determine pinch and grab separately
+            CheckIfJointPosesGrabbing();
+
             if (MRTKOculusConfig.Instance.UpdateMaterialPinchStrengthValue && handMaterial != null)
             {
+                if (IsGrabbing)
+                {
+                    pinchStrength = 1.0f;
+                }
                 handMaterial.SetFloat(pinchStrengthProp, pinchStrength);
             }
+
             return isTracked;
         }
 
+        protected void CheckIfJointPosesGrabbing()
```

Comment:
Ideally this would return a bool, which you set in the function this is called in, rather, than setting it in the nested state.
If any of the checks fail, the function returns false.

The way it is now, if the joints aren't tracked, it will never get set to false, which is what should happen.

Reply:
Looking at the function more carefully, it takes no arguments, and only if the requested joints are available will it do the computation and set IsGripping to the result.  That is intentional, and the way the original code was written - so that if there is a flicker in tracking, the grab is not automatically canceled.

Reply:
(If you look at how IsPinching works just above, it doesn't check whether tracking is good either; the release case explicitly does not check finger confidence.)

Reply:
> Ideally this would return a bool, which is used to update IsGripping in the function this is called in, rather, than setting it in the nested state.

The generalized function would actually have three potential cases
- yes, gripping
- no, not gripping
- I don't know, you decide what to do

So the generalized function shouldn't return a bool unless the caller is presumed to already have determined "I know" - which might not be solely determined by tracking / finger confidence outside the function, and that is why I wanted to make this function respect that third case by being a no-op in that case.